src/ikev2_parser.rs

```diff
@@ -31,16 +31,16 @@ named!(pub parse_ikev2_header<IkeV2Header>,
     )
 );
 
-pub fn parse_ikev2_payload_generic(i: &[u8]) -> IResult<&[u8],IkeV2GenericPayload> {
+named!(pub parse_ikev2_payload_generic<IkeV2GenericPayload>,
```

Comment:
Info: I prefer the `pub fn parse_ikev2_payload_generic` version, since it is easier to add documentation (and all `pub` functions should/will be documented)